package.json

```diff
@@ -36,7 +36,8 @@
     "*.toml": "pnpm run format:toml"
   },
   "engines": {
-    "pnpm": "7.25.0"
+    "pnpm": "8.2.0",
```

Comment:
Is there any particular reason you specify this version?

Reply:
@hyf0 As you tagged the PR, it would be a nice to have to be up to date with latest pnpm version and benefits from its latest features / improvements. 
Let me know if it causes any issue 
Thanks

Reply:
I wonder if we could specify a range of versions, such as `>=8.0.0`.

A specific version makes one pretty frustrated once you aren't using the version. You have to learn something like corepack to manipulate the version.

I'm going to wait for the decision of @hardfist. He is a very experienced developer in this aspect.

Reply:
Sure, in the mean time I updated the yaml files to target node 16 and updated the contribution doc as well
The CI passes now

Reply:
I don't think change engines.pnpm too often is a good practice,  it will require every developer to reinstall their global pnpm version, which is not necessary, and we only use very limited features of pnpm, so except pnpm upgrade bring feature| fix we really need, upgrade is not necessary.

Reply:
@hardfist thanks for the feedback 
I am not sure how often it was modified before - but in this case I completely agree apart from potential faster installation of packages thanks to https://github.com/pnpm/pnpm/pull/6281 which was released in `v7.30.4`  there is no critical reason to have this update. It is really just an nice to have as long as it does not disturb other contributors.
However in this same current PR, I also modified some `.yaml` files which were targeting node version 14 when the contribution documentation recommend node 16. Still nothing major, but good to have alignment.
If jugged non necessary / too risky I will close it no worries.
Thanks

Reply:
Hi @hyf0 
Looking at this change https://github.com/web-infra-dev/rspack/commit/99bec3fec54dbea969c157a8236938235ad1b8c2#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R39
In case contributors use different version of `pnpm`, the lock file might change unnecessary as you can see for instance in this PR on another repo https://github.com/TooTallNate/node-degenerator/pull/19 
Not sure its the best way for dev experience, what do you think ?

Reply:
I deeply agree with you we should use the latest pnpm version and benefits from its latest features / improvements.

We are considering switching to `corepack`, which would use a local package manager to override the global one while you under the `rspack` folder.

`corepack` would automatically install the correct package manager with the correct version.

That's how I think the process is going to be done:

1. introduce `corepack` in the rspack repo with the current pnpm version in another PR
2. update pnpm to 8.2.0, which is this PR does

We have a meeting today to discuss the process. I'll post the summary after the meeting.

Reply:
Noted thanks

Reply:
Sync up:

We have discussed this and going to introduce `corepack` in the Rspack repo. After this, The affection of updating the version of Pnpm would be reduced a lot.

We are also going to write a `setup` script to help contributors finish all setup work before running `pnpm install`.

I would mark this PR on hold and merge it after introducing `corepack`. Thank you in advance.

Reply:
thanks a lot for the update @hyf0